src/Routing/RouteCollection.php

```diff
@@ -147,6 +147,12 @@ public function parse(string $url, string $method = ''): array
 
         if (isset($this->staticPaths[$decoded])) {
```

Comment:
I just noticed, shouldn't this rather use a string with the possible query parameters removed?

Reply:
`$decoded` is the string after possible query params removed, that bit of code is not visible in the diff.

Reply:
The query parameters aren't removed from `$decoded` AFAICT tell, but from `$url`. I think the decoding part may need to be moved below the query parameters removal part.

Reply:
My bad, yes the query params are not removed from `$decoded`. I'll look into it.

That said we need to deprecate `RouterCollection::parse()` it's actually unused. `RoutingMiddleware` using `RouteCollection::parseRequest()`.

Reply:
Yeah, should've already been deprecated back in 3.x I guess. A test for both `parse()` and `parseRequest()` using a static path with query string arguments would be good though, it seems that the existing ones with using query strings are used with dynamic paths only.

Reply:
> Yeah, should've already been deprecated back in 3.x I guess. 

Yep I missed this back then. Lets get rid of it in 5.x. I'll take care of this.